File: 프로그래머스/2/17677. ［1차］ 뉴스 클러스터링/［1차］ 뉴스 클러스터링.py
def solution(str1, str2):
    answer = 0
    str1 = str1.lower()
    str2 = str2.lower()
    arr1 = []
    arr2 = []
    
    for i in range(len(str1) - 1):
        a = str1[i]
        b = str1[i+1]
        if a.isalpha() and b.isalpha():
            arr1.append(a+b)
    
    for i in range(len(str2) - 1):
        a = str2[i]
        b = str2[i+1]
        if a.isalpha() and b.isalpha():
            arr2.append(a+b)
    
    p = 0
    q = 0
    
    arr = arr1 + arr2
    arr = set(arr)
    
    for i in arr:
        if i in arr1 and i in arr2:
            p += min(arr1.count(i), arr2.count(i))
        q += max(arr1.count(i), arr2.count(i))
        
    if q == 0:
        answer = 65536
    else:
        answer = int(p/q*65536)
    
    
    
    return answer

File: 프로그래머스/2/17677. ［1차］ 뉴스 클러스터링/［1차］ 뉴스 클러스터링.cpp
#include <bits/stdc++.h>

using namespace std;

double a,b;
set<string> s;
vector<string> v1, v2;
string new_str1, new_str2;

int solution(string str1, string str2) {
    int answer = 0;
    if (str1.empty() && str2.empty()) return 1;

    for (int i=0;i<str1.size()-1;i++) {
        string temp1 = str1.substr(i, 2);

        int ok = 1;
        for (int j=0;j<temp1.size();j++) {
            if (!isalpha(temp1[j])) {
                ok = 0;
                break;
            }
            temp1[j] = tolower(temp1[j]);

        }
        if (ok) {
            v1.push_back(temp1);
            s.insert(temp1);
        }
    }

    for (int i=0;i<str2.size()-1;i++) {
        string temp2 = str2.substr(i, 2);

        int ok = 1;
        for (int j=0;j<temp2.size();j++) {
            if (!isalpha(temp2[j])) {
                ok = 0;
                break;
            }
            temp2[j] = tolower(temp2[j]);

        }
        if (ok) {
            v2.push_back(temp2);
            s.insert(temp2);
        }
    }


    for (string target : s) {
        int cnt1 = 0;
        int cnt2 = 0;

        for (string s1 : v1) {
            if (target == s1) cnt1++;
        }

        for (string s2 : v2) {
            if (target == s2) cnt2++;
        }

        a += min(cnt1, cnt2);
        b += max(cnt1, cnt2);
    }


    if (b == 0) return 65536;
    answer = a / b * 65536;

    return answer;
}